referenceframe/frame_system_test.go

```diff
@@ -237,3 +241,312 @@ func TestConvertTransformProtobufToFrameSystemPart(t *testing.T) {
+		outMap, err := FrameSystemToPCD(fs, nil, logger)
+		test.That(t, err, test.ShouldBeNil)
+
+		for i, v := range outMap["frame0"] {
+			test.That(t, spatial.R3VectorAlmostEqual(v, checkAgainst0[i], 1e-2), test.ShouldBeTrue)
+		}
+		for i, v := range outMap["frame1"] {
+			test.That(t, spatial.R3VectorAlmostEqual(v, checkAgainst1[i], 1e-2), test.ShouldBeTrue)
+		}
+	})
+
+	t.Run("incorrectly defined frame system, i.e. with nil parent for frame0", func(t *testing.T) {
+		fs := NewEmptySimpleFrameSystem("test")
+		logger := golog.NewTestLogger(t)
+		// ------
+		name := "frame"
+		pose := spatial.NewPoseFromPoint(r3.Vector{-4, -4, -4})
+		dims := r3.Vector{1, 1, 1}
+		geomCreator, err := spatial.NewBox(pose, dims, "box")
+		test.That(t, err, test.ShouldBeNil)
+		frame, err := NewStaticFrameWithGeometry(name, pose, geomCreator)
+		test.That(t, err, test.ShouldBeNil)
+		fs.AddFrame(frame, nil)
+		// -----
+		inputs := StartPositions(fs)
+		outMap, err := FrameSystemToPCD(fs, inputs, logger)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, outMap, test.ShouldBeEmpty)
+	})
+
+	t.Run("correct frame system with nil input and DOF = 0", func(t *testing.T) {
+		fs := NewEmptySimpleFrameSystem("test")
+		logger := golog.NewTestLogger(t)
+		// ------
+		name := "frame"
+		pose := spatial.NewPoseFromPoint(r3.Vector{-4, -4, -4})
+		dims := r3.Vector{1, 1, 1}
+		geomCreator, err := spatial.NewBox(pose, dims, "box")
+		test.That(t, err, test.ShouldBeNil)
+		frame, err := NewStaticFrameWithGeometry(name, pose, geomCreator)
+		test.That(t, err, test.ShouldBeNil)
+		fs.AddFrame(frame, fs.World())
+		// -----
+		outMap, err := FrameSystemToPCD(fs, nil, logger)
+		test.That(t, err, test.ShouldBeNil)
+		for i, v := range outMap["frame"] {
+			test.That(t, spatial.R3VectorAlmostEqual(v, checkAgainst0[i], 1e-2), test.ShouldBeTrue)
+		}
+	})
+
+	t.Run("correct frame system with a parent child relationship, with nil input and DOF != 0", func(t *testing.T) {
+		fs := NewEmptySimpleFrameSystem("test")
+		logger := golog.NewTestLogger(t)
+		jsonData, err := os.ReadFile(rdkutils.ResolveFile("config/data/model_frame_geoms.json"))
+		test.That(t, err, test.ShouldBeNil)
+		model, err := UnmarshalModelJSON(jsonData, "")
+		test.That(t, err, test.ShouldBeNil)
+		orientConf, err := spatial.NewOrientationConfig(spatial.NewZeroOrientation())
+		test.That(t, err, test.ShouldBeNil)
+		lc := &LinkConfig{
+			ID:          "arm",
+			Parent:      "world",
+			Translation: r3.Vector{1, 2, 3},
+			Orientation: orientConf,
+		}
+		lif, err := lc.ParseConfig()
+		test.That(t, err, test.ShouldBeNil)
+		// fully specified part
+		part := &FrameSystemPart{
+			FrameConfig: lif,
+			ModelFrame:  model,
+		}
+		armFrame, _, err := CreateFramesFromPart(part, logger)
+		test.That(t, err, test.ShouldBeNil)
+		fs.AddFrame(armFrame, fs.World())
+		// -----
+		outMap, err := FrameSystemToPCD(fs, nil, logger)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, outMap, test.ShouldBeEmpty)
```

Comment:
this is no longer a non-empty output due to the test above `t.Run("displaced box with another box as its child with nil inputs"`.

Reply:
In this test we also have a nil initial input for `inputs`, but one of the frames has DOF > 0.